packages/client/modules/teamDashboard/components/ProviderRow/ProviderRow.tsx

```diff
@@ -0,0 +1,197 @@
+  display: 'flex',
+  justifyContent: 'flex-start',
+  padding: Layout.ROW_GUTTER
+})
+
+interface Props {
+  connected: boolean
+  onConnectClick: () => void
+  connectButtonText?: string
+  connectButtonIcon?: React.ReactElement
+  submitting: boolean
+  togglePortal: () => void
+  menuRef: React.MutableRefObject<HTMLButtonElement | null> // TODO: make generic menu component
+  providerName: string
+  providerDescription: string
+  providerLogo: React.ReactElement
+  children?: React.ReactElement | false
+  seeHowItWorksUrl: string
+  contactUsProps?: {
+    url: string
+    onSubmit: () => void
+  }
+}
+
+const ProviderRow = (props: Props) => {
+  const {
+    connected,
+    onConnectClick,
+    connectButtonText = 'Connect',
+    connectButtonIcon = <Icon>add</Icon>,
+    submitting,
+    togglePortal,
+    menuRef,
+    providerName,
+    providerDescription,
+    providerLogo,
+    contactUsProps,
+    seeHowItWorksUrl,
+    children
+  } = props
+  const isDesktop = useBreakpoint(Breakpoint.SIDEBAR_LEFT)
+  return (
+    <ProviderCard>
+      <CardTop>
+        {providerLogo}
+        <RowInfo>
+          <ProviderName>{providerName}</ProviderName>
+          <RowInfoCopy>
+            {providerDescription}.{' '}
+            <a href={seeHowItWorksUrl} target='_blank' rel='noreferrer'>
+              See how it works
+            </a>
+            .
+          </RowInfoCopy>
+        </RowInfo>
+        <ProviderActions>
+          {!connected && !contactUsProps && (
+            <StyledButton
+              key='linkAccount'
+              onClick={onConnectClick}
+              palette='warm'
+              waiting={submitting}
+            >
+              {isDesktop ? connectButtonText : connectButtonIcon}
+            </StyledButton>
+          )}
+          {!connected && contactUsProps && (
+            <Form
+              method='get'
+              target='_blank'
+              action={contactUsProps.url}
+              onSubmit={contactUsProps.onSubmit}
+            >
+              <StyledButton key='request' palette='warm'>
+                {isDesktop ? 'Contact Us' : <IconOutlined>mail</IconOutlined>}
+              </StyledButton>
+            </Form>
+          )}
+          {connected && (
+            <>
+              {isDesktop ? (
+                <>
+                  <StatusWrapper>
+                    <StatusIcon>done</StatusIcon>
+                    <StatusLabel>Connected</StatusLabel>
+                  </StatusWrapper>
+                  <SmallMenuButton onClick={togglePortal} ref={menuRef}>
+                    <MenuSmallIcon>more_vert</MenuSmallIcon>
+                  </SmallMenuButton>
+                </>
+              ) : (
+                <MenuButton onClick={togglePortal} ref={menuRef}>
+                  <Icon>more_vert</Icon>
+                </MenuButton>
+              )}
+            </>
+          )}
+        </ProviderActions>
+      </CardTop>
+      {children}
```

Comment:
+2 great work making this reusable so we can save a ton of CSS in the other components!